### DISCUSSION TITLE
Nextcloud Sync not working anymore - ETag unquoting fails

### DISCUSSION THREAD

**Discussion by MrFloppy** [photoprism/photoprism#3496] [Q&A]
When starting the sync with nextcloud (27.0.0), photoprism throws the following error in logs:

`time="2023-06-22T09:02:52Z" level=error msg="service-folder: property <DAV: getetag>: webdav: failed to unquote ETag: invalid syntax"
`
Before that the sync worked 18 months without a challenge. Maybe it is nextcloud? Any ideas? Anybody else having problems here?
What else do you need from me in terms of information?

Below are nextcloud's response headers:

```
Cache-Control:
no-cache, no-store, must-revalidate
Content-Encoding:
gzip
Content-Length:
12345
Content-Security-Policy:
default-src 'none';base-uri 'none';manifest-src 'self';script-src 'nonce-b0xteTlMQmJ1M0JUd0R0T1lXeG1CaEtoTnBONWU5enVMU096ZlZBcmU1VT06eHZQRXcra3k5Z2NmcGtnTEZBQU5OU1RvUk1JYkVJbStibHZqU1RkSk1OYz0=';style-src 'self' 'unsafe-inline';img-src 'self' data: blob:;font-src 'self' data:;connect-src 'self';media-src 'self';frame-src 'self';frame-ancestors 'self';form-action 'self'

Content-Type:
text/html; charset=UTF-8
Date:
Thu, 22 Jun 2023 09:31:23 GMT
Expires:
Thu, 19 Nov 1981 08:52:00 GMT
Feature-Policy:
autoplay 'self';camera 'none';fullscreen 'self';geolocation 'none';microphone 'none';payment 'none'
Pragma:
no-cache
Referrer-Policy:
no-referrer
Server:
Apache/2.4.56 (Debian)
Strict-Transport-Security:
max-age=315360000; includeSubDomains; preload
X-Content-Type-Options:
nosniff
X-Frame-Options:
SAMEORIGIN
X-Permitted-Cross-Domain-Policies:
none
X-Powered-By:
PHP/8.2.7
X-Request-Id:
xhjmCh4xs94gW7HAynSm
X-Robots-Tag:
noindex, nofollow
X-Xss-Protection:
1; mode=block
```

**Comment by graciousgrey** [comment]
I converted this to a discussion until it is clear that this is a bug in PhotoPrism.

> What else do you need from me in terms of information?

--> You can find our bug template here: https://www.photoprism.app/kb/reporting-bugs

Have you changed anything in your setup recently? For example, have you updated Nextcloud or PhotoPrism? Are you using a proxy?

Is the sync really not working anymore or are you "just" getting these error messages?

**Comment by MrFloppy** [comment]
Sorry for leaving this out. 

The sync stopped working on April, 11th. Nextcloud still sees, that the app password gets used but no data gets synced.

Here we go:

1. What is not working as documented?
The sync shows "connection failed" when trying to execute the sync to nextcloud. Recreating app password on nextcloud side does not change this. Recreating the sync also does not change a thing. Connecting directly through WebDAV does work.

2. How can we reproduce it?
- Create app password in nextcloud
- Add new sync in photoprism with credentials and nextcloud url (e.g. example.com/nextcloud/remote.php/dav/files/USERNAME/
 or without "nextcloud" if using a subdomain

3. What behavior do you expect?
The sync synchronizes the "Instant-Upload" images to photorpism.

4. What could be the cause of your problem?
Maybe nextcloud, maybe traefik, maybe photoprism. All pieces are always updated to the latest versions.

5. Can you provide us with example files for testing, error logs, or screenshots?
Log output photoprism: 
`time="2023-06-24T19:21:21Z" level=error msg="service-folder: property <DAV: getetag>: webdav: failed to unquote ETag: invalid syntax"`

Log output nextcloud:
`XXX.XXX.XXX.XXX - XYZ [24/Jun/2023:19:20:57 +0000] "PROPFIND /remote.php/dav/files/XYZ/ HTTP/1.1" 207 5681 "-" "Go-http-client/2.0"
`


6. Which software versions do you use?
(a) PhotoPrism Architecture & Build Number: AMD64 Docker 230615

(b) Database Type & Version: MariaDB 10.5

(c) Operating System Types & Versions: Linux, Debian

(d) Browser Types & Versions: All ;)

(e) Ad Blockers, Browser Plugins, and/or Firewall Software? PiHole, but should not be the challenge here.



7. On what kind of device is PhotoPrism installed?

(a) Device / Processor Type: Ryzen 5 5600G

(b) Physical Memory & Swap Space in GB: 32GB RAM, 1GB SWAP

(c) Storage Type: SSD-Raid

(d) Anything else that might be helpful to know?

Not yet

8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?

traefik as reverse proxy 2.10.3


I hope that helps. What else do you need?

**Comment by graciousgrey** [comment]
Is someone else experiencing this?

I could not reproduce this (PhotoPrism [230615-90a18f6e7](https://docs.photoprism.app/release-notes/) Nextcloud 27.0.0). After setting up the sync, files have been downloaded from Nextcloud to PhotoPrism without problems.

I would assume this is most likely a configuration problem.

**Reply by bmschwa** [reply]
Yeah, I'm seeing this with Photoprism CE Build 240420-ef5f14bc4 w/ Nextcloud Hub 6 (v 27.1.9)

**Reply by lastzero** [reply]
What is this Hub thing? An extra software that runs with Nextcloud? What web server does it use?

**Reply by bmschwa** [reply]
I think "the hub thing" is just marketing for all the 'stuff' available in next cloud.

**Reply by bmschwa** [reply]
@lastzero, do you know if setting [PHOTOPRISM_LOG_LEVEL](https://docs.photoprism.app/getting-started/config-options/#logging) to DEBUG would output the response that the system isn't liking?

**Reply by EtienneBruines** [reply]
@lastzero To add some details to this (error message being the same). 

The XML response from the WebDAV server (from the `PROPFIND /`  request), contains entries like this:

```xml
    <D:response>
        <D:href>/Camera%20Upload/My%20Photos/Camera/IMG_20200812_081459244.jpg</D:href>
        <D:propstat>
            <D:prop>
                <D:resourcetype></D:resourcetype>
                <D:getcontentlength>3267942</D:getcontentlength>
                <D:getcontenttype>image/jpeg</D:getcontenttype>
                <D:getlastmodified>Wed, 12 Aug 2020 14:14:24 GMT</D:getlastmodified>
                <D:displayname>IMG_20200812_081459244.jpg</D:displayname>
                <D:getetag>0aa1cc5135610b957fa603bae09808a4d7ce04cf</D:getetag>
                <D:lockdiscovery/>
                <D:supportedlock>
                    <D:lockentry>
                        <D:lockscope>
                            <D:exclusive/>
                        </D:lockscope>
                        <D:locktype>
                            <D:write/>
                        </D:locktype>
                    </D:lockentry>
                    <D:lockentry>
                        <D:lockscope>
                            <D:shared/>
                        </D:lockscope>
                        <D:locktype>
                            <D:write/>
                        </D:locktype>
                    </D:lockentry>
                </D:supportedlock>
            </D:prop>
            <D:status>HTTP/1.1 200 OK</D:status>
        </D:propstat>
    </D:response>
```

I'm no expert as to what the error message `service-folder: property <DAV: getetag>: webdav: failed to unquote ETag: invalid syntax` actually means, but if it means "we tried to unquote the ETag and that didn't work", I can see why. The `D:getetag` is not (and probably doesn't have to be) quoted.

**Reply by EtienneBruines** [reply]
Might be somewhat related to this: https://github.com/emersion/go-webdav/pull/69 / https://github.com/emersion/go-webdav/issues/165
